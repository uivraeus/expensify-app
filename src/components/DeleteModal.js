import React from 'react';
import Modal from 'react-modal';

const DeleteModal = (props) => (
  <Modal 
    isOpen={props.queryConfirm}
    onRequestClose={props.onNo}
    contentLabel="Confirm deletion"
    closeTimeoutMS={200}
    className="modal"
  >
    <h3 className="modal__title">Do you really want to delete the expense?</h3>
    <div className="modal__button-group">
      <button className="button" onClick={props.onNo}>No</button>
      <button className="button" onClick={props.onYes}>Yes</button>
    </div>
  </Modal>
);

export default DeleteModal;