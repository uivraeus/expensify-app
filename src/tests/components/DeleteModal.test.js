import React from 'react';
import { shallow } from 'enzyme';
import DeleteModal from '../../components/DeleteModal';
import { wrap } from 'module';

let onYes, onNo, wrapper;
beforeEach(() => {
  onYes = jest.fn();
  onNo = jest.fn();
  wrapper = shallow(
    <DeleteModal
      queryConfirm={true} 
      onYes={onYes} 
      onNo={onNo} 
    />
  );
});

test('should render DeleteModal component with modal active', () => {
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('Modal').prop('isOpen')).toBe(true);
});

test('should handle No click', () => {
  const noBtn = wrapper.find('button').filterWhere(n => n.text()==='No');
  noBtn.simulate('click');
  
  expect(onNo).toHaveBeenCalled();
  expect(onYes).not.toHaveBeenCalled();  
});

test('should handle Yes click', () => {
  const noBtn = wrapper.find('button').filterWhere(n => n.text()==='Yes');
  noBtn.simulate('click');
  
  expect(onYes).toHaveBeenCalled();
  expect(onNo).not.toHaveBeenCalled();  
});