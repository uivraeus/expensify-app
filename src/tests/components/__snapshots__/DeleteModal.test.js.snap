// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DeleteModal component with modal active 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Confirm deletion"
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnOverlayClick={true}
>
  <h3
    className="modal__title"
  >
    Do you really want to delete the expense?
  </h3>
  <div
    className="modal__button-group"
  >
    <button
      className="button"
      onClick={[Function]}
    >
      No
    </button>
    <button
      className="button"
      onClick={[Function]}
    >
      Yes
    </button>
  </div>
</Modal>
`;
